### PR TITLE
add error test cases for transitive closure

### PR DESCRIPTION
**Pull request by kl52752**

### CODE REVIEW

File: pkg/cloud/rgraph/algo/trclosure/transitive_closure_test.go [hunk 1]
```diff
@@ -237,7 +239,76 @@ func TestTransitiveClosure(t *testing.T) {
 			},
 			want: wantGraph{ids: []string{"a", "b", "c"}},
 		},
-		// TODO(bowei): add error test cases.
+		{
+			name:  "don't traverse error nodes",
```
Comment (bowei) [inline pkg/cloud/rgraph/algo/trclosure/transitive_closure_test.go:243]: name: "nodes with error results in err"

File: pkg/cloud/rgraph/algo/trclosure/transitive_closure_test.go [hunk 2]
```diff
@@ -237,7 +239,76 @@ func TestTransitiveClosure(t *testing.T) {
...
+				addNode("b", []string{"d", "e"}, nodeIsInError)
+				addNode("d", []string{"a"})
+				addNode("e", []string{"b"})
+				return g
+			},
+			want:    wantGraph{ids: []string{"a", "b", "c"}},
+			wantErr: true,
+		},
+		{
+			name:  "don't traverse error nodes - child in error",
```
Comment (bowei) [inline pkg/cloud/rgraph/algo/trclosure/transitive_closure_test.go:257]: I think we need to define the behavior in the case where there is an error encountered. Do we expect that the traverse will continue and we get as much of the graph as we can OR will the traversal be canceled and the result be just whatever was fetched up around the time we detected the error and tried to stop?

If we can add this to the doc strings as well and also change the documented name of the test cases, this is will be more obvious to maintainers.

Comment (kl52752) [reply pkg/cloud/rgraph/algo/trclosure/transitive_closure_test.go:257]: yes, you are right there is no point checking the output on the error.

File: pkg/cloud/rgraph/algo/trclosure/transitive_closure_test.go (hunk 1 above)
Comment (kl52752) [reply pkg/cloud/rgraph/algo/trclosure/transitive_closure_test.go]: done

**Comment by kl52752** [comment]
/cc @mag-kol
//cc @tangenti

File: pkg/cloud/rgraph/algo/trclosure/transitive_closure_test.go [hunk 3]
```diff
@@ -237,7 +239,59 @@ func TestTransitiveClosure(t *testing.T) {
...
 		},
-		// TODO(bowei): add error test cases.
+		{
+			name:  "nodes with error results in err",
+			cloud: mockCloud,
+			graph: func() *rgraph.Builder {
+				g := rgraph.NewBuilder()
+				addNodeToGraph(g, "a", []string{"b", "c"})
+				addNode("b", []string{"d", "e"}, nodeIsInError)
+				addNode("d", []string{"a"})
```
Comment (bowei) [inline pkg/cloud/rgraph/algo/trclosure/transitive_closure_test.go:249]: Do we need these additional nodes if we are testing the error cases? We should make the error cases simple so people who maintain this later don't have to figure out whether or not the extra setup was part of the test? This comment applies to all of the test cases being added.

File: pkg/cloud/rgraph/rnode/fake/builder.go [hunk 4]
```diff
@@ -60,7 +62,7 @@ func (b *Builder) SetResource(u rnode.UntypedResource) error {
 func (b *Builder) SyncFromCloud(ctx context.Context, gcp cloud.Cloud) error {
 	// TODO(bowei): add ability to inject errors for the SyncFromCloud.
```
Comment (bowei) [inline pkg/cloud/rgraph/rnode/fake/builder.go:63]: Please fix this comment if it is no longer relevant.

File: pkg/cloud/rgraph/algo/trclosure/transitive_closure_test.go [hunk 5]
```diff
@@ -237,7 +239,40 @@ func TestTransitiveClosure(t *testing.T) {
...
 			want: wantGraph{ids: []string{"a", "b", "c"}},
 		},
-		// TODO(bowei): add error test cases.
+		{
+			name:  "nodes with error results in err",
+			cloud: mockCloud,
+			graph: func() *rgraph.Builder {
+				g := rgraph.NewBuilder()
+				addNodeToGraph(g, "a", []string{"b", "c"})
+				addNode("b", []string{"d", "e"}, nodeIsInError)
```
Comment (bowei) [inline pkg/cloud/rgraph/algo/trclosure/transitive_closure_test.go:248]: we need to fix these references to non-existant nodes? the same comment applies to all of the new test cases

Comment (kl52752) [reply pkg/cloud/rgraph/algo/trclosure/transitive_closure_test.go]: done

**Pull request merged**